recipes/google-cloud-cpp/all/conanfile.py

```diff
@@ -0,0 +1,279 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.28.0"
+
+class GoogleCloudCppConan(ConanFile):
+    name = "google-cloud-cpp"
+    description = "C++ Client Libraries for Google Cloud Services"
+    license = "Apache-2.0"
+    topics = "google", "cloud", "google-cloud-storage", "google-cloud-platform", "google-cloud-pubsub", "google-cloud-spanner", "google-cloud-bigtable"
+    homepage = "https://github.com/googleapis/google-cloud-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake", "cmake_find_package_multi", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False], 
+        "fPIC": [True, False]
+        }
+    default_options = {
+        "shared": False, 
+        "fPIC": True
+        }
+
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        
+        if self.settings.os == 'Windows' and self.options.shared:
```

Comment:
If this configuration is not working, why adding `set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)` in CMakeLists.txt?

Reply:
Well... I tried to too, and probably it is possible, but without a local Windows to test and try I gave up. Also, it will probably build for older VS versions... Someone will take care in another PR if needed 😊